pkg/antctl/antctl.go

```diff
@@ -669,6 +669,37 @@ $ antctl get podmulticaststats pod -n namespace`,
 			commandGroup:        get,
 			transformedResponse: reflect.TypeOf(agentapis.BGPPeerResponse{}),
 		},
+		{
+			use:     "bgproutes",
+			aliases: []string{"bgproute"},
+			short:   "Print the advertised bgp routes.",
+			long:    "Print the advertised bgp routes.",
+			example: `  Get the list of all advertised bgp routes
+  $ antctl get bgproutes
+  Get the list of advertised IPv4 bgp routes
+  $ antctl get bgproutes --ipv6=false
```

Comment:
This is a definite step back in terms of usability. The previous approach of `antctl get bgproutes --ipv4` made more sense to me. I think I kind of understand what the struggle is here, but I think there is still a way to solve it elegantly.

In antctl: `--ipv4` still means "ipv4 only", `--ipv6` still means "ipv6 only"
In the API server:
  - no query param means "both ip families"
  - reject URLs where the query params have values (e.g., `ipv4=true`), with a BadRequest error
  - reject URLs where both `ipv4` and `ipv6` are present, with a BadRequest error
  - if `ipv4` is present, only includes ipv4 routes
  - if `ipv6` is present, only includes ipv6 routes

This way antctl usage remains user-friendly, and the API is clear, well-defined and consistent

Reply:
Hi @antoninbas 
Keeping `ipv4` and `ipv6` bool flags in antctl and not `string=value` flag.

1. antctl get bgproutes --ipv4
   `r.URL.RawQuery` has value `ipv4=`
2. antctl get bgproutes --ipv4=true
    `r.URL.RawQuery` has value `ipv4=`
3. antctl get bgproutes --ipv4=false
    `r.URL.RawQuery` has value `""`
4. antctl get bgproutes
    `r.URL.RawQuery` has value `""`

Problem is how to distinguish between 3rd and 4th case ?

Reply:
You should think of `--ipv4` as `--ipv4-only` and `--ipv6` as `--ipv6-only`. If you prefer you can rename the flags as such.

So `antctl get bgproutes --ipv4=false` means `antctl get bgproutes --ipv4-only=false` and is strictly equivalent to `antctl get bgproutes`.

Renaming the flags will probably help avoid confusion. The flags will be a bit longer, but I expect that they won't be used too much.

Reply:
> So `antctl get bgproutes --ipv4=false` means `antctl get bgproutes --ipv4-only=false` and is strictly equivalent to `antctl get bgproutes`.

I don't think they are equivalent.
`antctl get bgproutes` will output both ipv4 and ipv6 routes.
`antctl get bgproutes --ipv4=false` should output only ipv6 routes ? but it will output both ipv4 and ipv6 which isn't correct.

Reply:
OK, then ipv4-only=false means output both. Fine.

Reply:
Lets keep `--ipv4-only` as mostly `--ipv4-only` will be used.  `--ip-only=false` will be least used and it is a bit pointless to use but at least if it is used we make sure its output is not wrong.